ai16z/eliza
# fix: Fix Parameter Parsing in plugin-evm TransferAction and Return Transaction Hash

# Relates to:

https://github.com/ai16z/eliza/issues/964

# Risks

Low – The changes affect the EVM plugin's transfer function, specifically the template parsing and parameter extraction. Other parts of the system should remain unaffected.

# Background

## What does this PR do?

- Fixes the parsing of transfer parameters in the `plugin-evm` module to ensure the `TransferAction` function receives the correct parameters.
- Improves the template parsing prompt for EVM transfers.
- Returns the transaction hash upon successful transfer execution.

## What kind of change is this?

- **Bug fix**: Resolves issues with parameter parsing preventing transfer execution.
- **Improvement**: Enhances the transfer template parsing and response handling to return transaction hashes.

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?

- Review changes in the EVM plugin’s `transfer` function and template parsing logic.

## Detailed testing steps

1. Configure an EVM wallet.
2. Ensure the wallet has sufficient ETH.
3. Issue a natural language command to perform a transfer (e.g., "Send 0.5 ETH to 0x123...").
4. Verify that the transfer is executed successfully and the transaction hash is returned.












## Discord username

0xtogo
